src/reader.rs

```diff
@@ -337,10 +337,19 @@ impl<'a> Reader<'a> {
             return Err(Error::UnsupportedCompVersion);
         }
 
-        if header.total_size as usize != blob.len() {
+        if header.total_size != blob.len() as u32
+            || header.total_size as usize != blob.len()
+        {
             return Err(Error::BadTotalSize);
         }
 
+        if header.total_size < header.struct_offset
```

Comment:
Can we merge these check with the previous ones, or you think it's clearer to keep them separately?

Reply:
I wasn't sure if another error variant would have been more appropriate than `BadTotalSize`, and always thought of these as somewhat separate checks. The new checks asserts that the header is intrinsically valid, the previous one asserts that the header describes the read buffer.

Reply:
Not sure as well if very granular errors needed to distinguish between the cases.